File: 04.Simulation模拟/leetcode_1861_rotateTheBox.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<char>> rotateTheBox(vector<vector<char>>& boxGrid) {
        int m = boxGrid.size(), n = boxGrid[0].size();
        for (int i=0; i<m; i++){
            int cnt = 0;
            for (int j=0; j<n; j++){
                char cur = boxGrid[i][j];
                if (cur == '#'){
                    cnt++;
                    boxGrid[i][j] = '.';
                }
                if (cur == '*'){
                    for (int k=0; k<cnt; k++){
                        boxGrid[i][j-k-1] = '#';
                    }
                    cnt = 0;
                }
                if (j == n-1){
                    for (int k=0; k<cnt; k++){
                        boxGrid[i][j-k] = '#';
                    }
                }
            }
        }
        vector<vector<char>> res(n, vector<char>(m));
        for (int i=0; i<m; i++){
            for (int j=0; j<n; j++){
                res[j][m-i-1] = boxGrid[i][j];
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<vector<char>> boxGrid = {{'#', '.', '#'}};
    vector<vector<char>> res = s.rotateTheBox(boxGrid);
    for (auto& i : res){
        for (auto& j : i){
            cout << j << " ";
        }
        cout << endl;
    }
}

File: 04.Simulation模拟/leetcode_1861_rotateTheBox.py
from typing import List

class Solution:
    def rotateTheBox(self, boxGrid: List[List[str]]) -> List[List[str]]:
        m, n = len(boxGrid), len(boxGrid[0])
        for i in range(m):
            cnt = 0
            for j in range(n):
                cur = boxGrid[i][j]
                if cur == "#":
                    cnt += 1
                    boxGrid[i][j] = '.'
                if cur == "*":
                    for k in range(cnt):
                        boxGrid[i][j-k-1] = '#'
                    cnt = 0
                if j == n-1:
                    for k in range(cnt):
                        boxGrid[i][j-k] = '#'
        
        res = [['']*m for _ in range(n)]
        for i in range(m):
            for j in range(n):
                res[j][m-i-1] = boxGrid[i][j]
        return res

if __name__ == "__main__":
    s = Solution()
    boxGrid = [["#",".","#"]]
    print(s.rotateTheBox(boxGrid))
